js/modules/k6/http/http.go

```diff
@@ -21,101 +21,152 @@
+	defaultClient *Client
+	exports       *goja.Object
+}
 
-// ErrJarForbiddenInInitContext is used when a cookie jar was made in the init context
-var ErrJarForbiddenInInitContext = common.NewInitContextError("Making cookie jars in the init context is not supported")
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
 
-// New returns a new global module instance
-func New() *GlobalHTTP {
-	return &GlobalHTTP{}
+// New returns a pointer to a new HTTP RootModule.
+func New() *RootModule {
+	return &RootModule{}
 }
 
-// GlobalHTTP is a global HTTP module for a k6 instance/test run
-type GlobalHTTP struct{}
-
-// NewModuleInstancePerVU returns an HTTP instance for each VU
-func (g *GlobalHTTP) NewModuleInstancePerVU() interface{} { // this here needs to return interface{}
-	return &HTTP{ // change the below fields to be not writable or not fields
-		TLS_1_0:                            netext.TLS_1_0,
-		TLS_1_1:                            netext.TLS_1_1,
-		TLS_1_2:                            netext.TLS_1_2,
-		TLS_1_3:                            netext.TLS_1_3,
-		OCSP_STATUS_GOOD:                   netext.OCSP_STATUS_GOOD,
-		OCSP_STATUS_REVOKED:                netext.OCSP_STATUS_REVOKED,
-		OCSP_STATUS_SERVER_FAILED:          netext.OCSP_STATUS_SERVER_FAILED,
-		OCSP_STATUS_UNKNOWN:                netext.OCSP_STATUS_UNKNOWN,
-		OCSP_REASON_UNSPECIFIED:            netext.OCSP_REASON_UNSPECIFIED,
-		OCSP_REASON_KEY_COMPROMISE:         netext.OCSP_REASON_KEY_COMPROMISE,
-		OCSP_REASON_CA_COMPROMISE:          netext.OCSP_REASON_CA_COMPROMISE,
-		OCSP_REASON_AFFILIATION_CHANGED:    netext.OCSP_REASON_AFFILIATION_CHANGED,
-		OCSP_REASON_SUPERSEDED:             netext.OCSP_REASON_SUPERSEDED,
-		OCSP_REASON_CESSATION_OF_OPERATION: netext.OCSP_REASON_CESSATION_OF_OPERATION,
-		OCSP_REASON_CERTIFICATE_HOLD:       netext.OCSP_REASON_CERTIFICATE_HOLD,
-		OCSP_REASON_REMOVE_FROM_CRL:        netext.OCSP_REASON_REMOVE_FROM_CRL,
-		OCSP_REASON_PRIVILEGE_WITHDRAWN:    netext.OCSP_REASON_PRIVILEGE_WITHDRAWN,
-		OCSP_REASON_AA_COMPROMISE:          netext.OCSP_REASON_AA_COMPROMISE,
+// NewModuleInstancePerVU returns an HTTP module instance for each VU.
+func (r *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	rt := vu.Runtime()
+	mi := &ModuleInstance{
+		vu:         vu,
+		rootModule: r,
+		exports:    rt.NewObject(),
+	}
+	mi.defineConstants()
 
+	mi.defaultClient = &Client{
+		// TODO: configure this from lib.Options and get rid of some of the
+		// things in the VU State struct that should be here.
+		moduleInstance:   mi,
 		responseCallback: defaultExpectedStatuses.match,
 	}
+
+	mustExport := func(name string, value interface{}) {
+		if err := mi.exports.Set(name, value); err != nil {
+			common.Throw(rt, err)
+		}
+	}
+
+	mustExport("url", mi.URL)
+	mustExport("CookieJar", mi.newCookieJar)
+	mustExport("cookieJar", mi.getVUCookieJar)
+	mustExport("file", mi.File) // TODO: deprecate or refactor?
```

Comment:
Does File need to stay exported?

Reply:
Resolved by https://github.com/grafana/k6/pull/2226/commits/d87b4d6721568af0d3d6a61ec6272d0ae3f7ccbe